orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -153,4 +176,10 @@ private static Writer newOrcWriter(OutputFile file,
                                                          createWriterFunc) {
     return (OrcRowWriter<D>) createWriterFunc.apply(schema, orcSchema);
   }
+
+  private TreeWriter treeWriterHiddenInORC() {
+    DynFields.BoundField<TreeWriter> treeWriterFiled =
+            DynFields.builder().hiddenImpl(writer.getClass(), "treeWriter").build(writer);
```

Comment:
Nit:  In iceberg we usually use 4 space indent, rather than 8 spaces.

Reply:
Another thing is:  the `treeWriter` is a private member in `writer`'s underlying implementation `WriterImpl`.  Is is possible to create a new pull request in [apache orc ](https://github.com/apache/orc) project to expose the `TreeWriter` or `estimateMemory()`  from `org.apache.orc.Writer` so that we can get rid of this in the further iceberg release ? 

CC @rdblue @dongjoon-hyun @marton-bod @pvary